applications/ContactStructuralMechanicsApplication/custom_linear_solvers/mixedulm_linear_solver.h

```diff
@@ -403,6 +403,22 @@ class MixedULMLinearSolver :
         VectorType& rB
         ) override
     {
+        // We print the system before condensate (if needed)
+        if (mEchoLevel == 2) { //if it is needed to print the debug info
+            KRATOS_INFO("RHS BEFORE CONDENSATION") << "RHS  = " << rB << std::endl;
+        } else if (mEchoLevel == 3) { //if it is needed to print the debug info
+            KRATOS_INFO("LHS BEFORE CONDENSATION") << "SystemMatrix = " << rA << std::endl;
+            KRATOS_INFO("RHS BEFORE CONDENSATION") << "RHS  = " << rB << std::endl;
+        } else if (mEchoLevel >= 4) { //print to matrix market file
+            std::stringstream matrix_market_name;
+            matrix_market_name << "before_condensation_A_" << mFileCreated << ".mm";
+            TSparseSpaceType::WriteMatrixMarketMatrix((char *)(matrix_market_name.str()).c_str(), rA, false);
```

Comment:
Is the explicit conversion to `char*` necessary? I thought this is already the output of `c_str()`?

btw you could directly use a string:
`const std::string matrix_market_name = "before_condensation_A_" + std::to_string(mFileCreated) + ".mm";`

and then
`TSparseSpaceType::WriteMatrixMarketMatrix((char *)(matrix_market_name).c_str(), rA, false);`
or even
`TSparseSpaceType::WriteMatrixMarketMatrix((matrix_market_name).c_str(), rA, false);`

Reply:
I assume I copied the code from somewhere, I will update the things

Reply:
Done